config/initializers/5_telemetry_notifications.rb

```diff
@@ -0,0 +1,27 @@
+telemetry = Foreman::Telemetry.instance
+
+ActiveSupport::Notifications.subscribe(/process_action.action_controller/) do |*args|
+  event = ActiveSupport::Notifications::Event.new(*args)
+  controller = event.payload[:controller].underscore
+  action = event.payload[:action].underscore
+  status = event.payload[:status]
+
+  telemetry.increment_counter(:http_requests, 1, :controller => controller, :action => action, :status => status)
+  telemetry.observe_histogram(:http_request_total_duration, event.duration || 0, :controller => controller, :action => action)
+  telemetry.observe_histogram(:http_request_db_duration, event.payload[:db_runtime] || 0, :controller => controller, :action => action)
+  telemetry.observe_histogram(:http_request_view_duration, event.payload[:view_runtime] || 0, :controller => controller, :action => action)
+
+  # measure GC stats for each request
+  before = Thread.current[:foreman_telemetry_gcstats]
+  after = GC.stat
+  GC_METRICS.each do |ruby_key, metric_name|
+    telemetry.increment_counter(metric_name, after[ruby_key] - before[ruby_key], :controller => controller, :action => action) if after.include?(ruby_key)
+  end
+end if telemetry.enabled?
+
+ActiveSupport::Notifications.subscribe(/instantiation.active_record/) do |*args|
+  event = ActiveSupport::Notifications::Event.new(*args)
+  class_name = event.payload[:class_name]
+  record_count = event.payload[:record_count]
+  telemetry.increment_counter(:activerecord_instances, record_count, :class => class_name)
+end if telemetry.enabled?
```

Comment:
Could the code in this file live under `lib/foreman/telemetry` ? If this is required to be initialized here, could the code live there but be a one line call out to it? I think it might be more obvious code wise to have everything under `lib/foreman/telemetry` witha simple call here like `telemetry.notifications`